shacl-af/index.html

```diff
@@ -651,230 +644,101 @@ <h3>SPARQL-based Target Types</h3>
   sh:property [sh:path ex:politics; sh:in ("Democrat" "Republican"); sh:minCount 1; sh:maxCount 1];
-  sh:property [sh:path ex:politician; sh:node ex:PoliticianRefShape; sh:minCount 1];
 .</pre>
-        </section>
-        <section id="performance-considerations">
-          <h4>Performance Considerations</h4>
-          <p>WARNING: selecting nodes by shape can be a very expensive operation, since it requires a shape engine to examine a potentially large number of nodes.</p>
-          <ul>
-            <li>
-              Target shapes that use node characteristics not usually indexed by RDF repositories can be especially expensive.
-              This includes selecting by URL pattern (e.g. see <a href="#target-namespace-instances">Namespace Instances</a> above).
-              by literal characteristic (e.g. see <a href="#target-all-langstrings">All langStrings</a> above),
-              by literal value length
-              (<a href="https://www.w3.org/TR/shacl/#MinLengthConstraintComponent">sh:minLength</a>, 
-              <a href="https://www.w3.org/TR/shacl/#MaxLengthConstraintComponent">sh:maxLength</a>), etc
-            </li>
-            <li>
-              Target shapes that use properties and objects to describe nodes can typically be executed faster,
-              using the indexes of the RDF repository.
-              This includes targeting <a href="#target-by-property-and-object">by property and object</a>,
-              <a href="#target-by-cardinality">by cardinality</a>,
-              <a href="#semantic-type-discrimination">"semantic type"</a>, etc.
-            </li>
-          </ul>
-          <p>In addition, the population size of the nodes to be examined has a great impact:</p>
-          <ul>
-            <li>Targeting by shape will be slow for "whole repository" validation, where potentiall every node needs to be examined.</li>
-            <li>In contrast, targeting by shape is feasible for validating the payload of a transaction, which may be significantly smaller than a whole repository.</li>
-          </ul>
-          <p>
-            Please check with your SHACL validation engine provider whether <code>NodeShape</code> targets can be calculated efficiently.
-            For example, the <a href="https://rdf4j.org/javadoc/latest/org/eclipse/rdf4j/sail/shacl/ShaclSail.html">RDF4J ShaclSail</a>
-            implements incremental validation of transactions and works efficiently for some typical data update scenarios.
-          </p>
-        </section>
-		  </section>
+        <div class="def def-text">
+					<div class="def-header">TEXTUAL DEFINITION</div>
+					<div class="def-text-body">
+						If triple <code>S sh:targetShape TS</code> is in the <a>shapes graph</a>
+						then the <a>target</a> nodes of <code>S</code> are all nodes (<a>subjects</a> or <a>objects</a>) in the <a>data graph</a>
+            that <a>conform</a> to <code>TS</code>.
```

Comment:
Linked **nodes** to definition.

I'd like to keep it because the mechanism works on nodes and is neutral re subjects or objects. A node will be targeted no matter whether it's a subject, object or both.

RDF graphs have no nodes except by virtue of triples, but I think that's an unfortunate artefact: people think of nodes as the first ingredient of graphs.